src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -50,58 +53,50 @@ The interesting case is when the characteristic of `F` is odd.
 
 section quadratic_char
 
+open mul_char
+
 variables {F : Type*} [field F] [fintype F] [decidable_eq F]
 
 /-- Some basic API lemmas -/
-lemma quadratic_char_eq_zero_iff (a : F) : quadratic_char F a = 0 ↔ a = 0 :=
+lemma quadratic_char_eq_zero_iff (a : F) : quadratic_char' F a = 0 ↔ a = 0 :=
```

Comment:
It think it is better to state this using `quadratic_char`. The idea is that lemmas about `quadratic_char'` (or `quadratic_char_def`) exist only to make the definition of `quadratic_char` better, and that the corresponding result for `quadratic_char` follows immediately since it is multiplicative or whatever. Once `quadratic_char` is defined, nobody should never use `quadratic_char'`.

Also, in `iff` statement we usually use implicit variables (but we are not very strict about this).

Reply:
To avoid having to `change` the goal into something in terms of `quadratic_char_def`, I'm renaming this to `quadratic_char_eq_zero_iff'`; then I can add `quadratic_char_eq_zero_iff`, whose proof is just `quadratic_char_eq_zero_iff'`.